index.bs

```diff
@@ -8083,21 +8079,395 @@ objects.
+        1.  Return.
+    1.  Let |index| be 0.
+    1.  Let |result| be a [=list=] containing |total| null values.
+    1.  [=list/For each=] |promise| of |promises|:
+        1.  Let |promiseIndex| be |index|.
+        1.  Let |fulfillmentHandler| be the following steps:
+            1.  Let |arg| be the first argument to these steps.
+            1.  Set |result|[|promiseIndex|] to |arg|.
+            1.  Set |fullfilledCount| to |fullfilledCount| + 1.
+            1.  If |fullfilledCount| equals |total|, then perform |successSteps| given |result|.
+        1.  Let |fulfillmentHandler| be [=!=] [$CreateBuiltinFunction$](|fulfillmentHandler|, « »):
+        1.  Perform [$PerformPromiseThen$](|promise|, |fulfillmentHandler|, |rejectionHandler|).
+        1.  Set |index| to |index| + 1.
+</div>
+
+This phrase is useful when you wish to aggregate the result of multiple promises, and react to them
+all together, in the same way that {{Promise/all()|Promise.all()}} functions for JavaScript code.
+
+<div algorithm>
+    To
+    <dfn id="waiting-for-all-promise" export lt="get a promise to wait for all|getting a promise to wait for all">get a promise for waiting for all</dfn>
+    with a [=list=] of <a interface lt="Promise">Promise&lt;|T|&gt;</a> values |promises| and a
+    [=Realm=] |realm|, perform the following steps:
+
+    1.  Let |promise| be [=Promise/a new promise=] of type Promise&lt;sequence&lt;|T|>> in |realm|.
+    1.  Let |successSteps| be the following steps, given |results|:
+        1.  [=Promise/Resolve=] |promise| with |results|.
+    1.  Let |failureSteps| be the following steps, given |reason|:
+        1.  [=Promise/Reject=] |promise| with |reason|.
+    1.  [=Promise/Wait for all=] with |promises|, given |successSteps| and |failureSteps|.
+    1.  Return |promise|.
+</div>
+
+This phrase is useful when you wish to aggregate the results of multiple promises, and then produce
+another promise from them.
+
+
+<h5 id="es-promise-examples">Examples</h5>
+
+<div class="example" id="promise-example-delay">
+
+    <code>delay</code> is a function that returns a promise that will be fulfilled in a number of
+    milliseconds. It illustrates how simply you can resolve a promise, with one line of prose.
+
+    <pre highlight="webidl">
+        interface I {
+          Promise&lt;void> delay(unrestricted double ms);
+        };
+    </pre>
+
+<div algorithm="delay">
+
+    The <code>delay(|ms|)</code> method steps are:
+
+    1.  Let |realm| be <b>this</b>'s [=relevant Realm=].
+    1.  If |ms| is NaN, let |ms| be +0; otherwise let |ms| be the maximum of |ms| and +0.
+    1.  Let |p| be [=Promise/a new promise=] in |realm|.
+    1.  Run the following steps [=in parallel=]:
+        1.  Wait |ms| milliseconds.
+        1.  [=Promise/Resolve=] |p| with the {{void}} value.
+    1.  Return |p|.
+</div>
+
+The equivalent function in JavaScript would be
+
+<pre highlight="js">
+function delay(ms) {
+    // Verify the this value is correct.
```

Comment:
```suggestion
    // Omitted: verify the this value is correct.
```

and the same below.

Although, I am unsure if we really need to preserve these JavaScript versions, given the Web IDL spec audience. They are somewhat problematic, depending on how strongly you take "equivalent".